substrate/frame/support/procedural/tools/src/lib.rs

```diff
@@ -50,10 +50,24 @@ pub fn generate_crate_access(unique_id: &str, def_crate: &str) -> TokenStream {
 	}
 }
 
+/// Check if a path is using the `frame` crate or not.
+///
+/// This will usually check the output of [`generate_crate_access_2018`].
+pub fn is_using_frame_crate(path: &syn::Path) -> bool {
+	path.segments.first().map(|s| s.ident == "frame").unwrap_or(false)
```

Comment:
why checking just `.first()`? 🤔 possibly my own old cold, but can't understand now.

Reply:
yeah normally you would do `last()` here in case it is in some weird re-export / path

Reply:
We want to know if the `path` is reexported from `frame` (something like `frame::deps::xyz`) or not. Not if the path takes us to `frame`, and that's why we don't use `last()`.
We could also check if the `frame` crate itself was reexported by checking all the elements in `segments`, but I don't think that's the use case. I will add a comment to the fn to clarify this